core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -262,15 +267,22 @@ private[spark] class ExecutorAllocationManager(
     val maxNeeded = maxNumExecutorsNeeded
 
     if (maxNeeded < numExecutorsTarget) {
-      // The target number exceeds the number we actually need, so stop adding new
-      // executors and inform the cluster manager to cancel the extra pending requests
-      val oldNumExecutorsTarget = numExecutorsTarget
-      numExecutorsTarget = math.max(maxNeeded, minNumExecutors)
-      client.requestTotalExecutors(numExecutorsTarget)
-      numExecutorsToAdd = 1
-      logInfo(s"Lowering target number of executors to $numExecutorsTarget because " +
-        s"not all requests are actually needed (previously $oldNumExecutorsTarget)")
-      numExecutorsTarget - oldNumExecutorsTarget
+      if (!numTargetExecutorAdjustable.get) {
+        // Keep the initial number of target executor to not ramp down until the first job is
+        // submitted or the first idle executor is released.
+        client.requestTotalExecutors(numExecutorsTarget)
+        0
+      } else {
+        // The target number exceeds the number we actually need, so stop adding new
+        // executors and inform the cluster manager to cancel the extra pending requests
+        val oldNumExecutorsTarget = numExecutorsTarget
+        numExecutorsTarget = math.max(maxNeeded, minNumExecutors)
+        client.requestTotalExecutors(numExecutorsTarget)
```

Comment:
@sryza and @andrewor14 , do we need to avoid requesting executors also here and `addExecutors` when `oldNumExecutorsTarget` == `numExecutorsTarget`, not in initializing status?